File: src/main/java/a/b/noride2/Utils/FunctionUtils.java
package a.b.noride2.Utils;

import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.player.Player;

public class FunctionUtils {
    public static void processRiddenEntities(Player player, EntityProcessor processor) {
        Entity currentEntity = player;
        while (currentEntity != null) {
            // 执行对当前实体的处理逻辑
            processor.process(currentEntity);

            // 获取当前实体的乘客（即当前实体所骑乘的实体）
            currentEntity = currentEntity.getVehicle();
        }
    }
}

File: src/main/kotlin/a/b/noride2/utils/function/FunctionUtils.kt
package a.b.noride2.utils.function

import net.minecraft.world.entity.Entity
import com.github.cecnull1.cecnull1lib.utils.InfixFunction.nbtIn

infix fun Entity.inEntityPersistentNbtAndRemove(
    key: String
): Boolean {
    val persistentData = this.persistentData
    val hasKey = persistentData nbtIn key
    val keyValue = if (hasKey) persistentData.getBoolean(key) else false

    // 检查键是否存在且值为 true
    if (hasKey && keyValue) {
        // 清除指定键
        persistentData.remove(key)
        return true
    }
    return false
}

infix fun Entity.inEntityPersistentNbt(key: String) : Boolean {
    val persistentData = this.persistentData
    val hasKey = persistentData nbtIn key
    return hasKey
}